components/TARGET_PSA/services/attestation/COMPONENT_PSA_SRV_IMPL/attest_crypto.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2019, Arm Limited. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ */
+
+#include "t_cose/src/t_cose_crypto.h"
+#include "tfm_plat_defs.h"
+#include "crypto.h"
+#include "tfm_plat_crypto_keys.h"
+#include <string.h>
+
+static psa_hash_operation_t hash_handle;
+
+enum t_cose_err_t
+t_cose_crypto_pub_key_sign(int32_t cose_alg_id,
+                           int32_t key_select,
+                           struct useful_buf_c hash_to_sign,
+                           struct useful_buf signature_buffer,
+                           struct useful_buf_c *signature) {
+    enum t_cose_err_t cose_ret = T_COSE_SUCCESS;
+    psa_status_t crypto_ret;
+
+    (void)key_select;
+
+    const psa_key_id_t key_id = 17;
+    psa_key_handle_t handle = 0;
+
+    crypto_ret = psa_crypto_init();
+    if (crypto_ret != PSA_SUCCESS)
+    {
+        return T_COSE_ERR_HASH_GENERAL_FAIL;
+    }
+
+    crypto_ret = psa_open_key(PSA_KEY_LIFETIME_PERSISTENT, key_id, &handle);
+    if (crypto_ret != PSA_SUCCESS)
+    {
+        return T_COSE_ERR_NO_KID;
```

Comment:
should we cleanup init if we fail?
mbedtls_psa_crypto_free( );?

Reply:
I removed all calling to ```psa_crypto_init```  - we assume init called before crypto operations, if not an error will be return.